compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DataflowAnalyzer.java

```diff
@@ -278,11 +280,18 @@ public static DataflowAnalyzer getInstance(CompilerContext context) {
      */
     public BLangPackage analyze(BLangPackage pkgNode) {
         this.uninitializedVars = new LinkedHashMap<>();
+        this.unusedErrorVarsDeclaredWithVar = new HashMap<>();
         this.globalNodeDependsOn = new LinkedHashMap<>();
         this.functionToDependency = new HashMap<>();
         this.dlog.setCurrentPackageId(pkgNode.packageID);
         SymbolEnv pkgEnv = this.symTable.pkgEnvMap.get(pkgNode.symbol);
         analyzeNode(pkgNode, pkgEnv);
+
+        for (Map.Entry<BSymbol, Location> entry : this.unusedErrorVarsDeclaredWithVar.entrySet()) {
+            this.dlog.error(entry.getValue(), DiagnosticErrorCode.UNUSED_VARIABLE_WITH_INFERRED_TYPE_INCLUDING_ERROR,
+                            entry.getKey().name);
+        }
+
```

Comment:
Can we move this to `visit(BLangPackage pkgNode)`

Reply:
Fixed in 415c3aa.